### PR TITLE
Fix Users error on SAR dashboard 

### PR DESCRIPTION
**Pull request by keeysnc**
### Description
This should fix the SAR error banner from temporarily showing up on the SAR dashboard when a user's WP has been approved.

Screenshot 2024-04-23 at 12 44 37 PM

### Related ticket(s)

CMDCT-[https://jiraent.cms.gov/browse/CMDCT-3423]

---
### How to test
1. Fill out WP as state user
2. Approved WP as admin
3. Once report is approved go back in as a state user and view the SAR dashboard
4. The SAR error banner should not be enabled.

### Notes

---
### Pre-review checklist

- [ ] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
- [ ] I have performed a self-review of my code
- [ ] I have manually tested this PR in the deployed cloud environment

---
#### Review
- [ ] Design: This work has been reviewed and approved by design, if necessary
- [ ] Product: This work has been reviewed and approved by product owner, if necessary

#### Security
_If either of the following are true, notify the team's ISSO (Information System Security Officer)._

- [ ] These changes are significant enough to require an update to the SIA.
- [ ] These changes are significant enough to require a penetration test.
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

File: services/ui-src/src/components/pages/Dashboard/DashboardPage.tsx [hunk 1]
```diff
@@ -92,6 +92,9 @@ export const DashboardPage = ({ reportType }: Props) => {
   const [selectedReport, setSelectedReport] = useState<AnyObject | undefined>(
     undefined
   );
+  const [wpToCopyFrom, setWorkPlanToCopyFrom] = useState<boolean>(
```
Comment (jessabean) [inline services/ui-src/src/components/pages/Dashboard/DashboardPage.tsx:95]: So, there is already a function to set the WP to copy from in the Report Provider. Which makes this state a little redundant.

I was thinking, maybe we could instead move where we currently call `setWorkPlanToCopyFrom` [in the fetch call](https://github.com/Enterprise-CMCS/macpro-mdct-mfp/blob/main/services/ui-src/src/components/reports/ReportProvider.tsx#L116). Like maybe move that line to an `else` block on [line 138](https://github.com/Enterprise-CMCS/macpro-mdct-mfp/blob/main/services/ui-src/src/components/reports/ReportProvider.tsx#L138)?

Comment (keeysnc) [reply services/ui-src/src/components/pages/Dashboard/DashboardPage.tsx]: Good catch, I had to look through it before commenting on this but I think the undefined might be causing the issue.  `fetchReportForSarCreation(activeState)` is being called in DashboardPage (line 119) and it's not resolving right away which I think is returning an `undefined` at first which is causing the banner to temporarily flicker. 

I added a null as the default value so even if fetchReportForSarCreation doesn't resolve right away the banner wont render until a null or `workPlanToCopyFrom` object comes through. 

Open to suggestions of this solution though.

**Review by gmrabian — APPROVED** [review]
After approving as an admin I checked the SAR dashboard and the banner did appear for a second before disappearing.
When I logged in as a state user it didn't show up at all.

Looks good! I don't think the flicker I saw on admin is a huge concern, just calling out that the state isn't cleared right away

File: services/ui-src/src/components/pages/Dashboard/DashboardPage.tsx [hunk 2]
```diff
@@ -293,16 +293,18 @@ export const DashboardPage = ({ reportType }: Props) => {
...
-          <Alert
-            title={sarVerbiage.alertBanner.title}
-            showIcon={true}
-            icon={alertIcon}
-            status={AlertTypes.ERROR}
-            description={sarVerbiage.alertBanner.body}
-            sx={sx.alertBanner}
-          />
-        )}
+        {reportType === ReportType.SAR &&
```
Comment (jessabean) [inline services/ui-src/src/components/pages/Dashboard/DashboardPage.tsx:296]: This is just a nit, but when the conditional gets this long, I usually like to pull it into a variable, just to keep the template a bit cleaner.

**Review by jessabean — DISMISSED** [review]
Smol nit, but this is working as expected!

File: services/ui-src/src/components/pages/Dashboard/DashboardPage.tsx (hunk 2 above)
Comment (keeysnc) [reply services/ui-src/src/components/pages/Dashboard/DashboardPage.tsx]: Sounds good, just updated this with a variable for the logic.

**Review by jessabean — APPROVED** [review]

**Review by gmrabian — COMMENTED** [review]
When I create a SAR the error pops up. I understand that I am now locked from creating another, but from a UX perspective it looks like something went wrong by creating it

Screenshot 2024-04-29 at 11 18 59 AM

Update: Spoke with Megan and she verified this is the correct flow. Possible circle back to better the UX for this.

**Review by gmrabian — APPROVED** [review]

**Pull request merged**